docs/proposals/17-shoot-control-plane-migration-bad-case.md

```diff
@@ -0,0 +1,102 @@
+
+9. After the `etcd-backup-restore` process in the destination seed detects that a final full snapshot exists, it copies all snapshots (or a subset of all snapshots) from the source to the destination backup container. When this is done, the Job finishes successfully which signals to the reconciliation flow that the snapshots have been copied.
+
+   **Note:** To save time, only the final full snapshot taken in step 6, or a subset defined by some criteria, could be copied, instead of all snapshots.
+
+10. The special "copy" version of the `etcd-main` Etcd resource is deleted from the source seed, and as a result the `etcd-copy` Job is also deleted by `etcd-druid`.
+
+11. The additional "source" backup entry referencing the source seed backup container is deleted from the Garden cluster and the destination seed. As a result, its corresponding `source-etcd-backup` secret is also deleted from the destination seed.
+
+12. From this point, the reconciliation flow proceeds as already described in [GEP-7](07-shoot-control-plane-migration.md). This is safe, since the source seed cluster is no longer able to interfere with the shoot.
+
+## Handling Inability to Access the Backup Container
+
+The mechanism described above assumes that the `etcd-backup-restore` process in the source seed is able to access its backup container in order to take snapshots. If this is not the case, but an ownership change was detected, the `etcd-backup-restore` process still sets the readiness probe status of the main `etcd` container to 503, and kills the main `etcd` process as described above to ensure that any open connections from `kube-apiserver` are terminated. This effectively deactivates the source seed cluster to ensure that the ownership of the shoot can be passed to a different seed.
+
+Because of this, `etcd-backup-restore` process in the destination seed responsible for copying the snapshots can avoid waiting forever for a final full snapshot to appear. Instead, after a certain timeout has elapsed, it can proceed with the copying. In this situation, whatever latest snapshot is found in the source backup container will be restored in the destination seed. The shoot is still migrated to a healthy seed at the cost of losing the etcd data that accumulated between the point in time when the connection to the source backup container was lost, and the point in time when the source seed cluster was deactivated.
+
+When the connection to the backup container is restored in the source seed, a final full snapshot will be eventually taken. Depending on the stage of the restoration flow in the destination seed, this snapshot may be copied to the destination seed and restored, or it may simply be ignored since the snapshots have already been copied.
```

Comment:
Why? Shouldn't the backup container detect that the ownership has changed after the connection is restored?

Reply:
It would detect the change of ownership, but it wouldn't be able to take the snapshot until the connection is back. Once it's back, it will still take the final snapshot. This snapshot might end up unused, but it doesn't really matter. It should still be taken because in some situation it might be used, and it's rather difficult to make the source seed aware of the state of the restoration flow in the destination seed to distinguish between both cases.